Multi archives

frivoal opened the pull request:
This change adds the ability to declare multiple archives of the same type (tag, category, year, month, or day) each using a specific permalink and layout, while maintaining compatibility with existing syntax.

The design of the feature is such that people who do not need this functionality can ignore its existence, and those who do can easily opt into it.

My personal use for this is that I want to generate two different view of the tags archive: one that is just an index of posts for each tag, with title and links, and one that inlines all the posts for that tag in a single page.

This pull request includes updates to the documentation and relevant additions to the test suite.

frivoal commented:
I've updated to the latest master, resolved conflicts, and made rubocop happy again. Feedback appreciated.

renshuki commented:
+1 for this feature.
I think it's also useful in case you have a multilingual Jekyll blog.

frivoal commented on History.markdown:
```diff
@@ -1,5 +1,9 @@
 ## HEAD
 
+### Major Enhancements
```
Fixed.

frivoal commented:
rubocop didn't like a two letter parameter, so I renamed id to key.

It is still unhappy about one thing: class TestJekyllArchives < Minitest::Test has too many lines. Should I just split that class into several ones? That doesn't sound all that useful to me, but if that's required, why not...

ashmaroli commented:
@frivoal I think you took my review comment a bit too literally. When I suggested defining private helpers `:basic_helper_method` and `:advanced_helper_method`, I was only giving examples.. :laughing: 
Sorry, I wasn't clear on that.

Please rename the helper methods to something that denotes what they do.. :smiley:

ashmaroli commented:
@frivoal The `master` branch has progressed ahead significantly since you proposed these changes.
If you're still interested in this feature and would like to contribute, I invite you to submit a fresh PR based of the latest `master`.
Resolving the conflicts on the branch is a non-trivial task.
Sorry for rejecting this one.
